### DISCUSSION TITLE
Add environment-dependant removeConsole compiler transform

### DISCUSSION THREAD

**Discussion by juliocorzo** [vercel/next.js#36686] [Ideas]
### Describe the feature you'd like to request

Babel provided the ability to remove console only on production, which is valuable when you're trying to diagnose errors in a staging environment:

```js
{
  "presets": ["next/babel"],

  "env": {
    "production": {
      "plugins": ["transform-remove-console"]
    }
  }
}
```

The same transformer introduced in 12.1 doesn't seem to have this feature. I'd appreciate if something like this was added.

### Describe the solution you'd like

Allow us to pass an `environments` config into the compiler options:

```ts
    removeConsole?:
      | boolean
      | {
          exclude?: string[]
          environment?: string[]
        }
```

Where doing this:

```js
// next.config.js
module.exports = {
  compiler: {
    removeConsole: {
      environment: ['production'],
    },
  },
}
```

Would remove console logs from production.

### Describe alternatives you've considered

A solution is switching back to Babel, or removing the console logs, none of which is really a good idea.

**Comment by juliocorzo** [comment]
A possible workaround if this isn't implemented is setting an environment variable to conditionally remove logs:

```js
// next.config.js
module.exports = {
  compiler: {
    removeConsole: process.env.REMOVE_CONSOLE_LOGS,
  },
}
```

**Comment by icyJoseph** [comment]
Hi,

Wouldn't this suffice?

```js
/** @type {import('next').NextConfig} */
const nextConfig = {
  reactStrictMode: true,
  compiler: {
    removeConsole: process.env.NODE_ENV === "production",
  },
};

module.exports = nextConfig;
```

**Reply by juliocorzo** [reply]
Absolutely, but I also think this should be a feature of the compiler and also documented, as many people moving from Babel might encounter the same issue and have the same question.

**Reply by icyJoseph** [reply]
I see, well, as a third party, my personal opinion is that the API as is, is good enough. 

It is shown here, https://nextjs.org/docs/advanced-features/compiler#remove-console, and there shouldn't be any expectation that the same old API ought to be reimplemented. Refining the API it is part of migrating and improving.

The `next.config.js` file contains just about everything you need to control the `removeConsole` API as a function of the environment, or even other variables/conditions. And the `removeConsole` as a boolean with exclusion is enough.
